_specs/ecip-tabs.md

```diff
@@ -0,0 +1,301 @@
+---
+ecip: 110x
+title: TD*TABS Consensus Arbitration
+author: Isaac (https://github.com/whilei)
+discussions-to: TODO
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-04-20
+requires: EIP-100, ECIP-1105
+lang: en
+---
+
+# Abstract
+
+Use a heuristic of chain state capitalization as an augmentive factor in consensus arbitration to improve normal finality rates and increase the cost of private chain production.
+
+# Motivation
+
+## State Finality Is Low and Slow
+
+![](https://i.imgur.com/c2biCLD.png)
+
+This proposal intends to increase the cost of producing a private, competitive chain segment, which is characteristic of double-spend attacks. It aims to improve finality expectations.
+
+It aims to maintain all existing assumptions and incentives associated with the incumbent proof-of-work protocol.
+
+# Specification
+
+This specification is defined as a modification of the general,[^1] currently existing canon-arbitration algorithm, namely "Modified GHOST," currently in use on Ethereum and Ethereum Classic networks.
+
+[^1]: Difficulty bomb and/or delay features are ignored.
+
+## TABS: Total Active Balance Synthesis
+
+Total Active Balance Synthesis (TABS) is a positive scalar integer. Generally considered, it provides a measure of currency capitalization over time for some chain state.
+
+It is defined to approach the median aggregate balance of all transaction-originating accounts for a blockchain of some length. It is denominated in Wei.
+
+TABS is a "synthetic" value derived from a "raw" Total Active Balance (TAB) value of any block and, recursively, the TABS value of its parent block, much like block difficulty and gas limit.
+
+### TAB: Total Active Balance
+
+The TAB value of a block is the sum of the balances (at the parent block state) of all unique transaction senders in the current block.
+
+Pseudo-code is provided as:
+
+```go
+// blockTab returns the "raw" TAB value for a block.
+// This value is the sum of the balances (at the parent state) 
+// of all transaction senders in this block.
+func blockTab(block *Block) (tab *big.Int) {
+    
+    // uniqueSenders is a lookup to de-dupe sender accounts.
+    // Each sender account only gets counted once.
+    var uniqueSenders = make(map[common.Address]bool)
+    
+    // Iterate all transactions in the block
+    for _, tx := range block.transactions {
+        var _, seen = uniqueSenders[tx.sender]
+        if !seen { 
+            uniqueSenders[tx.sender] = true
+            tab += state.balance(tx.sender, block.parentHash)    
+        }
+    }    
+    return tab
```

Comment:
where is the miner balance?

Reply:
Miner balance is gone. It was a "convenient shortcut" idea for miners originally, but I've removed it entirely now. The aggregated balance MUST represent guaranteed, "proven" balance. Tallying the balance of a block's `coinbase` is not a proven, active balance. 

An attacker could, by modifying the coinbase of block they produce, derive a greater balance than actually available to them.